src/hooks/use-update-demo-site.tsx

```diff
@@ -32,7 +32,7 @@ export const DemoSiteUpdateProvider: React.FC< DemoSiteUpdateProviderProps > = (
 				// No-op if logged out
 				return;
 			}
-			setDemoSiteUpdating( true );
+			setUpdatingSites( ( prev ) => new Set( prev ).add( localSite.id ) );
```

Comment:
The current implementation is good, so no need to change it. But wanted to share an alternative in case we'd like to use a string array for the state:

```suggestion
			setUpdatingSites( ( prev ) => [ ...new Set( [ ...prev, localSite.id ] ) ] );
```